docs/source/features/parallelisms.rst

```diff
@@ -80,10 +72,36 @@ The distributed optimizer in NeMo is built on top of
 `DistributedFusedAdam <https://github.com/NVIDIA/apex/blob/master/apex/contrib/optimizers/distributed_fused_adam.py>`_
 from Apex.
 
+Fully-Shared Data Parallelism (FSDP)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+NeMo supports Fully-Sharded Data Parallelism (FSDP) that shards parameter gradients and low-precision parameters for computation on top of the model states that Distributed optimizer shards (optimizer states and high-precision parameters).
+Since FSDP shards the entire model states, it ensures linear model state memory saving with increasing DP size.
+FSDP can be preferred for the LLM training with unbalanced workload between pipeline stages (or Transformer layers) or with a large vocabulary size, where pipelining would cause huge computation bubbles due to the workload imbalance.
+Also, FSDP unloads the effort to search the performance-optimal mappings with 3D parallelism (TP/PP/DP) because it has a single parallelization domain.
+
+NeMo uses `pytorch's FSDP interface <https://pytorch.org/tutorials/intermediate/FSDP_tutorial.html>`_ to shard LLM model states, which flattens the parameters of each Transformer layer and partitions across datap-parallel GPUs.
+FSDP introduces collectives across data-parallel GPUs; all-gather of the parameters for computation and reduce-scatter of parameter gradients.
```

Comment:
FSDP introduces collective communication across data-parallel GPUs. Specifically, it involves an all-gather operation for collecting parameters during computation and a reduce-scatter operation for distributing parameter gradients.